firstvoices/backend/views/contact_us_views.py

```diff
@@ -0,0 +1,174 @@
+import logging
+import re
+from smtplib import SMTPException
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.core.mail import send_mail
+from django.core.validators import validate_email
+from drf_spectacular.utils import (
+    OpenApiResponse,
+    extend_schema,
+    extend_schema_view,
+    inline_serializer,
+)
+from rest_framework import mixins, serializers
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+
+from backend.models import AppJson, Site
+from backend.serializers.contact_us_serializers import ContactUsSerializer
+from backend.utils.contact_us_utils import get_fallback_emails
+from backend.views import doc_strings
+from backend.views.api_doc_variables import site_slug_parameter
+from backend.views.base_views import SiteContentViewSetMixin
+from backend.views.doc_strings import error_403
+from backend.views.exceptions import ContactUsError
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="Returns the list of receiver emails used by the contact us post endpoint.",
+        responses={
+            200: ContactUsSerializer,
+            403: OpenApiResponse(description="Todo: Not authorized for this Site"),
+            404: OpenApiResponse(description="Todo: Site not found"),
+        },
+        parameters=[site_slug_parameter],
+    ),
+    create=extend_schema(
+        description="Sends emails to the addresses listed on the site contact_us_email and contact_us_users fields.",
+        responses={
+            202: OpenApiResponse(description="Success. Email successfully sent."),
+            403: OpenApiResponse(description=doc_strings.error_403_site_access_denied),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+        request=inline_serializer(
+            name="Contact Us Form",
+            fields={
+                "name": serializers.CharField(),
+                "email": serializers.CharField(),
+                "message": serializers.CharField(),
+            },
+        ),
+    ),
+)
+class ContactUsView(
+    SiteContentViewSetMixin,
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    GenericViewSet,
+):
+    http_method_names = ["get", "post"]
+    serializer_class = ContactUsSerializer
+    pagination_class = None
+
+    def get_queryset(self):
+        return self.get_validated_site()
+
+    def list(self, request, *args, **kwargs):
+        """
+        Returns the list of receiver emails used by the contact us post endpoint.
+        """
+
+        site = self.get_validated_site().first()
+        perm = Site.get_perm("change")
```

Comment:
This is a good choice for the permission, but please add a comment about it as well since it is different than our usual model permissions